openedx/core/djangoapps/course_groups/cohorts.py

```diff
@@ -343,25 +348,19 @@ def add_user_to_cohort(cohort, username_or_email):
         ValueError if user already present in this cohort.
     """
     user = get_user_by_username_or_email(username_or_email)
-    previous_cohort_name = None
-    previous_cohort_id = None
 
-    course_cohorts = CourseUserGroup.objects.filter(
-        course_id=cohort.course_id,
-        users__id=user.id,
-        group_type=CourseUserGroup.COHORT
-    )
-    if course_cohorts.exists():
-        if course_cohorts[0] == cohort:
-            raise ValueError("User {user_name} already present in cohort {cohort_name}".format(
-                user_name=user.username,
-                cohort_name=cohort.name
-            ))
-        else:
-            previous_cohort = course_cohorts[0]
-            previous_cohort.users.remove(user)
-            previous_cohort_name = previous_cohort.name
-            previous_cohort_id = previous_cohort.id
+    membership = CohortMembership(course_user_group=cohort, user=user)
+    membership.save()
+
+    """
+    TODO: with the new setup, we cannot emit a "requested" event containing
+    information about the previous cohort before calling save().
+
+    Decide how to best handle this
+        -remove previous info from "requested" event
+        -wait until after save so previous information is available
```

Comment:
My main concern arose after I wrote these inline comments and found this code (https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/course_groups/cohorts.py#L38), which seems to be emitting an "action happened" event, in contrast to "the "action requested" event here. Could this timing information be getting gathered and utilized elsewhere? If that's the case, our best course of action would be to move the `previous_cohort` information to that event, and continue to emit the "requested" event prior to attempting the save.

Who would be the best person to ask about this design? @explorerleslie? someone from Analytics?